RPA.MIT.Notification.Function/Notification.cs

```diff
@@ -3,143 +3,146 @@
 
             try
             {
                 var isValid = ValidateMessage.IsValid(notificationMsg);
 
                 if (!isValid)
                 {
-                    log.LogError("Invalid message: {notificationMsg}", notificationMsg);
-                    return;
+                    _logger.LogError("Invalid message: {notificationMsg}", notificationMsg);
                 }
 
                 dynamic notificationMsgObj = JObject.Parse(notificationMsg);
                 string templateName = notificationMsgObj.Action;
                 string scheme = notificationMsgObj.Scheme;
-                var templateId = _configuration[$"templates:{templateName}"];
-                var emailAddress = _configuration[$"schemas:{scheme}"];
+                var templateId = _configuration[$"templates{templateName}"];
+                var emailAddress = _configuration[$"schemas{scheme}"];
                 string id = notificationMsgObj.Id;
 
                 if (templateId == null)
                 {
-                    log.LogError("Template not found for action: {action}", templateName);
+                    _logger.LogError("Template not found for action: {action}", templateName);
                     await _eventQueueService.CreateMessage(id, "failed", "notification", "Template not found", notificationMsg);
-                    return;
                 }
 
                 if (emailAddress == null)
                 {
-                    log.LogError("emailAddress not found for scheme: {scheme}", scheme);
+                    _logger.LogError("emailAddress not found for scheme: {scheme}", scheme);
                     await _eventQueueService.CreateMessage(id, "failed", "notification", "emailAddress not found", notificationMsg);
-                    return;
                 }
 
-                log.LogInformation("Sending email for incoming message id: {id}", id);
+                _logger.LogInformation("Sending email for incoming message id: {id}", id);
 
                 var notifyResponse = _notifyService.SendEmail(emailAddress, templateId, notificationMsgObj.Data);
+
+                _logger.LogInformation("Sent email for incoming message id: {id}", id);
+
                 await _eventQueueService.CreateMessage(id, "sent", "notification", "Email sent", notificationMsg);
-                var notificationEntity = new NotificationEntity()
+
+                _logger.LogInformation("Sent queue message for incoming message id: {id}", id);
+
+                await _notificationTable.Add(new NotificationEntity()
                 {
                     PartitionKey = id,
                     RowKey = Guid.NewGuid().ToString(),
                     Status = "sent",
                     NotifyId = notifyResponse.id,
                     RetryCount = 0,
                     Data = notificationMsg
-                };
-                await tableClient.AddEntityAsync(notificationEntity);
+                });
```

Comment:
You may want to add .ConfigureAwait(false) to the async Add to avoid deadlocks.

Reply:
Code removed.